docs/releases.md

```diff
@@ -1,39 +1,157 @@
+- **Release Stage 4 - Complete the Release** - Finalize the release, start the next release.
+
+The Release Stages are not linked to Release Candidate numbers, in fact, there can be multiple release candidate per stages as we catch bugs and improve the release itself.
+
+<p align="center">
+  <a href="https://ipfs.io">
+    <img src="https://gateway.ipfs.io/ipfs/QmaFtLxoCAm5vFQ9AftKkhJwSAdDdF1jzV9DfzW6gbXqFL/Paper.Sketches.23.png" width="450" />
+  </a>
+</p>
+
+## Performing a Release
+
+The first step is for the `Lead Maintainer` for `go-ipfs` to open an issue with Title `go-ipfs <version> Release` and a c&p of the following template:
+
+```
+> <short tl;dr; of the release>
+
+# 🗺 What's left for release
+
+<List of items with PRs and/or Issues to be considered for this release>
+
+# 🔦 Highlights
+
+<List of items with PRs and/or Issues to be considered for this release>
+
+# 🏗 API Changes
+
+<List of API changes, if any>
+
+# ✅ Release Checklist
+
+For each RC published in each stage:
 - [ ] version string in `version.go` has been updated
 - [ ] tag commit with vX.Y.Z-rcN
 
-## Pre-Release Checklist
-- [ ] before release, tag 'release candidate' for users to test against
-  - if bugs are found/fixed, do another release candidate
-- [ ] all tests pass (no exceptions)
-- [ ] run interop tests https://github.com/ipfs/interop#test-with-a-non-yet-released-version-of-go-ipfs
-- [ ] webui works (for most definitions of 'works') - Test the multiple pages and verify that no visible errors are shown.
+### Release Stage 1 - Internal testing
+
+When Release Stage 1, there is a features freeze for the release branch.
+
 - [ ] CHANGELOG.md has been updated
   - use `./bin/mkreleaselog` to generate a nice starter list
-- [ ] version string in `repo/version.go` has been updated
-- [ ] tag commit with vX.Y.Z
-- [ ] update release branch to point to release commit (`git merge vX.Y.Z`).
-- [ ] publish dist.ipfs.io
-- [ ] publish next version to https://github.com/ipfs/npm-go-ipfs
-
-## Post-Release
-- [ ] Bump version string in `repo/version.go` to `vX.Y.Z-dev`
-- [ ] Upload the final release to the github releases page: https://github.com/ipfs/go-ipfs/releases
-- Communication
-  - [ ] Create the release issue
-  - [ ] Announcements (both pre-release and post-release)
-    - [ ] Twitter
-    - [ ] IRC
-    - [ ] Reddit
-  - [ ] Blog post (at minimum, paste the changelog. optionally add context and thank contributors.)
-- [ ] Update HTTP-API Documentation on the Website using https://github.com/ipfs/http-api-docs
+- [ ] Automated Testing - Ensure that all tests are passing, this includes:
+  - [ ] unit
+  - [ ] sharness
+  - [ ] [interop](https://github.com/ipfs/interop#test-with-a-non-yet-released-version-of-go-ipfs)
+- [ ] Network Testing:
+  - [ ] test lab things
+- [ ] Infrastructure Testing:
+  - [ ] Deploy new version to a subset of Bootstrappers
```

Comment:
Can we even do this yet? cc @ipfs/wg-infrastructure.

Reply:
@mburns ?